include/cute/atom/copy_traits_xe.hpp

```diff
@@ -1,80 +1,266 @@
+  // Reference map from (thr,val) to bit
+  using RefLayout = DstLayout;
+  // 32 bits register file
+  using CopyInternalType = uint;
+
+  GTensor tensor;
+};
+
+template <class GTensor>
+struct Copy_Traits<XE_2D_U16X16X16X2X1_LD_N, GTensor>
+     : XE_2D_LD_Unpack<XE_2D_U16X16X16X2X1_LD_N>
+{
+  // // using ThrID   = Layout<_16>; //TODO: I think it should be 16 (copy is
+  // per subgroup) - but static_assert fails
+  using ThrID = Layout<_16>;
+  // Map from (src-thr,src-val) to bit
+  using SrcLayout = Layout<Shape<_16, _64>, Stride<_0, _1>>;
+  // Map from (dst-thr,dst-val) to bit
+  using DstLayout =
+      Layout<Shape<_16, Shape<_16, _32>>, Stride<_32, Stride<_512, _1>>>;
+  // Reference map from (thr,val) to bit
+  using RefLayout = DstLayout;
+  // 32 bits register file
+  using CopyInternalType = uint;
+
+  GTensor tensor;
+};
+
+template <class CopyOp>
+struct XE_2D_ST_Unpack
+{
+    template <class TS, class SLayout,
+              class TD, class DLayout>
+    CUTE_HOST_DEVICE friend constexpr void
+    copy_unpack(Copy_Traits const &traits,
+                Tensor<TS, SLayout> const &src,
+                Tensor<ViewEngine<ArithmeticTupleIterator<TD>>, DLayout> &dst)
     {
-        // using ThrID   = Layout<_16>; //TODO: I think it should be 16 (copy is per subgroup) - but static_assert fails
-        using ThrID = Layout<_1>;
-        using NumBits = Int<sizeof(typename GTensor::engine_type::value_type) * 8>; // hacky: does vec of 8
-        // Map from (src-thr,src-val) to bit
-        using SrcLayout = Layout<Shape<_1, NumBits>>; // TODO:  is _1 correct?
-        // Map from (dst-thr,dst-val) to bit
-        using DstLayout = Layout<Shape<_1, NumBits>>;
-        // Reference map from (thr,val) to bit
-        using RefLayout = SrcLayout;
-
-        GTensor tensor;
-
-        template <class TS, class SLayout,
-                  class TD, class DLayout>
-        CUTE_HOST_DEVICE friend constexpr void
-        copy_unpack(Copy_Traits const &traits,
-                    Tensor<TS, SLayout> const &src,
-                    Tensor<ViewEngine<ArithmeticTupleIterator<TD>>, DLayout> &dst)
-        {
-            static_assert(is_rmem<TS>::value);
-            int H = size<0>(traits.tensor);
-            int W = size<1>(traits.tensor) * sizeof(typename decltype(traits.tensor)::engine_type::value_type);
-            auto [y, x] = dst.data().coord_;
-            XE_2D_SAVE::copy(traits.tensor.data().get(), W, H, W, int2_{x, y}, &*src.data());
-        }
-    };
-
-    template <class Copy, class GEngine, class GLayout>
-    auto make_xe_2d_copy(Tensor<GEngine, GLayout> gtensor)
+        static_assert(is_rmem<TS>::value);
+        int H = size<0>(traits.tensor);
+        int W = size<1>(traits.tensor) * sizeof(typename decltype(traits.tensor)::engine_type::value_type);
+        auto [y, x] = dst.data().coord_;
+        CopyOp::copy(traits.tensor.data().get(), W, H, W, int2_{x, y}, &*src.data());
```

Comment:
We should cast these coordinates to `int` in order to avoid the compile time error below. Note that this error arises when calling copy from the cutlass PVC pipeline and not through the cute PVC sample.
```suggestion
        CopyOp::copy(traits.tensor.data().get(), W, H, W, int2_{static_cast<int>(x), static_cast<int>(y)}, &*src.data());
```

`
error: non-constant-expression cannot be narrowed from type 'std::tuple_element<0, const std::tuple<unsigned long, unsigned long>>::type' (aka 'const unsigned long') to 'int' in initializer list [-Wc++11-narrowing]
   67 |             XE_2D_SAVE::copy(traits.tensor.data(), W, H, W, int2_{x, y}, &*src.data());
      |                                                                      ^
/home/tanvir/work/cutlass/include/cute/atom/copy_traits_xe.hpp:67:70: note: insert an explicit cast to silence this issue
   67 |             XE_2D_SAVE::copy(traits.tensor.data(), W, H, W, int2_{x, y}, &*src.data());
      |                                                                      ^
      |                                                                      static_cast<int>( )
`